observer.go

```diff
@@ -39,50 +43,27 @@ type observer struct {
 	observers []Observer
 }
```

Comment:
when old-style observers are passed to tracer or config via option, they are immediately converted to the contrib-style, thus we never need to use the old style observer, it is fully replaced by the new `compositeObserver`